waspc/src/Generator/WebAppGenerator.hs

```diff
@@ -0,0 +1,83 @@
+module Generator.WebAppGenerator
+       ( generateWebApp
+       ) where
+
+import Data.Aeson (ToJSON(..), (.=), object)
+import qualified Path
+import Path ((</>), reldir, relfile)
+
+import qualified Util
+import qualified Path.Aliases as Path
+import CompileOptions (CompileOptions)
+import Wasp
+import Generator.FileDraft
+import Generator.ExternalCodeGenerator (generateExternalCodeDir)
+import qualified Generator.WebAppGenerator.EntityGenerator as EntityGenerator
+import qualified Generator.WebAppGenerator.PageGenerator as PageGenerator
+import qualified Generator.WebAppGenerator.ButtonGenerator as ButtonGenerator
+import qualified Generator.WebAppGenerator.Common as C
+import qualified Generator.WebAppGenerator.ExternalCodeGenerator as WebAppExternalCodeGenerator
+
+
+generateWebApp :: Wasp -> CompileOptions -> [FileDraft]
+generateWebApp wasp _ = concatMap ($ wasp)
+    [ (:[]) . generateReadme
```

Comment:
wow what magic is this, to put in the array right?

Reply:
Yes, `(:[])` is same as `\x -> x:[]` or `\x -> [x]`.
It uses "section" feature of Haskell, where you can do stuff like `(+3)` or `(5-)` and similar, as long as we are dealing with a binary operator.